docker/docker-compose.dev.yml

```diff
@@ -1,5 +1,5 @@
 # Docker Compose file for development
-version: "2"
+version: "3.4"
```

Comment:
This is the lowest version of docker-compose file format that supports build targets. We could go higher if we only want to support recent versions of docker (this affects the version that contributors have installed [for .dev.] and whatever is installed on jenkins [for .test.]).

Reply:
We're running docker 18.06 in production.

```
param@frank:~$ docker --version
Docker version 18.06.0-ce, build 0ffa825
param@frank:~$ 
```

Jenkins has the following
```
Docker environment:
Docker version 1.12.3, build 6b644ec
docker-compose version 1.23.2, build 1110ad0
```

I'd be happy to go as up as this supports.

Reply:
It looks like we'll have to upgrade docker on jenkins then (according to https://docs.docker.com/compose/compose-file/#compose-and-docker-compatibility-matrix docker 1.12 only supports compose-file 2.1)

Reply:
Even though we're using docker 1.12 in jenkins, it's connecting to a 18.03 server, which is why the tests for this PR succeeded. However, we can upgrade jenkins anyway: https://github.com/metabrainz/jenkins/pull/1

It looks like we have no problem with updating this minimum version.